workspaces/config/lib/index.js

```diff
@@ -523,22 +523,33 @@ class Config {
       }
     }
 
-    const typeDesc = typeDescription(type)
-    const oneOrMore = typeDesc.indexOf(Array) !== -1
+    const typeDesc = typeDescription(type);
     const mustBe = typeDesc
-      .filter(m => m !== undefined && m !== Array)
-    const oneOf = mustBe.length === 1 && oneOrMore ? ' one or more'
-      : mustBe.length > 1 && oneOrMore ? ' one or more of:'
-      : mustBe.length > 1 ? ' one of:'
-      : ''
-    const msg = 'Must be' + oneOf
+      .filter(m => m !== undefined && m !== Array);
+    const msg = 'Must be' + this.getOneOfKeywords(mustBe,typeDesc);
     const desc = mustBe.length === 1 ? mustBe[0]
-      : mustBe.filter(m => m !== Array)
-        .map(n => typeof n === 'string' ? n : JSON.stringify(n))
-        .join(', ')
+      : [...new Set(mustBe.map(n => typeof n === 'string' ? n : JSON.stringify(n)))].join(', ');
     log.warn('invalid config', msg, desc)
   }
 
+  getOneOfKeywords (mustBe, typeDesc) {
+    let keyword;
+    switch (true) {
```

Comment:
We don't have too many switch statements and tend not to make new ones.  A series of early returning `if` statements would be more appropriate here to be inline with how we do things elsewhere.

Reply:
@wraithgar So, should I change this logic to `if` statements instead of `switch`?